providers/gce/gce_loadbalancer.go

```diff
@@ -279,7 +279,10 @@ func (g *Cloud) EnsureLoadBalancerDeleted(ctx context.Context, clusterName strin
 }
 
 func getSvcScheme(svc *v1.Service) cloud.LbScheme {
-	if t := GetLoadBalancerAnnotationType(svc); t == LBTypeInternal {
+	if LBTypeInternal == GetLoadBalancerAnnotationType(svc) {
+		return cloud.SchemeInternal
+	}
+	if hasFinalizer(svc, ILBFinalizerV1) {
```

Comment:
FinalizerV1 is always an InternalLoadbalancer ?

Reply:
yes, it's ***ILB***FinalizerV1

btw ILBFinalizerV2 is for subsettoing

Reply:
lol, I miss the I on ILB

Reply:
so , it should be if `if hasFinalizer(svc, ILBFinalizerV1)  !!  hasFinalizer(svc, ILBFinalizerV2` ?

Reply:
V2 is for ingress-gce controller. if we trigger deletion here this controller will remove nothing

Reply:
ILBFinalizerV2 is handled by ingress-gce

Reply:
but I am not sure we should execute the deletion then...
in ensureInternalLoadBalancerDeleted we don't skip for ILB subsetting services

and if the LB has annotation we will call ensureInternalLoadBalancerDeleted
if it does not we will call the ensureExternalLoadBalancerDeleted which seems much worse